### PR TITLE
chore: enable OSSF Scorecard analysis

### PR DESCRIPTION
**Pull request by fgreinacher**
Let's give this a try here to see whether it provides helpful insights.

Set up as described at https://github.com/marketplace/actions/ossf-scorecard-action

### CODE REVIEW

File: .github/workflows/scorecard.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+name: Scorecard supply-chain security
+on:
+  # For Branch-Protection check. Only the default branch is supported. See
+  # https://github.com/ossf/scorecard/blob/main/docs/checks.md#branch-protection
+  branch_protection_rule:
+  # To guarantee Maintained check is occasionally updated. See
+  # https://github.com/ossf/scorecard/blob/main/docs/checks.md#maintained
+  schedule:
+    - cron: '23 13 * * 3'
```
Comment (nejch) [inline .github/workflows/scorecard.yml:13]: Should we also allow a manual run here to test it out/do quick fixes?

```suggestion
  schedule:
    - cron: '23 13 * * 3'
  workflow_dispatch:
```

BTW, very specific cron there :rofl:

**Review by nejch — APPROVED** [review]
LGTM just one suggestion to enable manual runs if we need to tweak it but happy to merge as is.

I remember trying this out on python-gitlab and it was pretty pedantic, but let's give it a shot :bow:

File: .github/workflows/scorecard.yml (hunk 1 above)
Comment (fgreinacher) [reply .github/workflows/scorecard.yml:13]: Yes, good idea!

**Comment by fgreinacher** [comment]
> LGTM @fgreinacher just one suggestion to enable manual runs if we need to tweak it but happy to merge as is.
> 
> I remember trying this out on python-gitlab and it was pretty pedantic, but let's give it a shot 🙇

Thanks @nejch! Yes, I agree, it should not be too painful so I wanted to start with a tiny repo. Let's see :)

**Comment by fgreinacher** [comment]
@nejch It's quite OK I'd say: https://github.com/siemens/.github/security/code-scanning :)

**Pull request merged**